src/adaptation/precond.jl

```diff
@@ -2,389 +2,227 @@
 #### Robust online (co-)variance estimators.
 ####
 
-## Variance estimator
+abstract type MassMatrixAdaptor <: AbstractAdaptor end
 
-abstract type VarEstimator{T} end
+initialize!(adaptor::T, n_adapts::Int) where {T<:MassMatrixAdaptor} = nothing
+finalize!(adaptor::T) where {T<:MassMatrixAdaptor} = nothing
 
-# NOTE: this naive variance estimator is used only in testing
-mutable struct NaiveVar{E<:AbstractVecOrMat{<:AbstractFloat},T<:AbstractVector{E}} <: VarEstimator{T}
-    n :: Int
-    S :: T
-end
+## Unit mass matrix adaptor
 
-NaiveVar(::Type{T}, ::Tuple{Int}) where {T<:AbstractFloat} = NaiveVar(0, Vector{Vector{T}}())
-NaiveVar(::Type{T}, ::Tuple{Int,Int}) where {T<:AbstractFloat} = NaiveVar(0, Vector{Matrix{T}}())
+struct UnitMassMatrix{T<:AbstractFloat} <: MassMatrixAdaptor end
 
-function add_sample!(nc::NaiveVar, s::AbstractVecOrMat)
-    nc.n += 1
-    push!(nc.S, s)
-end
+Base.show(io::IO, ::UnitMassMatrix) = print(io, "UnitMassMatrix")
 
-function reset!(nc::NaiveVar{T}) where {T}
-    nc.n = 0
-    nc.S = T()
-end
+UnitMassMatrix(::Type{T}=Float64) where {T} = UnitMassMatrix{T}()
```

Comment:
As suggested above, IMO it would be more natural to define
```suggestion
UnitMassMatrix() = UnitMassMatrix{Float64}()
```
and just call `UnitMassMatrix{T}()` explicitly if possible.

Reply:
IMO that makes it clearer that `T` is the actual type parameter.